bin/gbuild

```diff
@@ -119,16 +119,24 @@ def build_one_configuration(suite, arch, build_desc, reference_datetime)
       script.puts "GBUILD_COMMON_CACHE=$HOME/cache/common"
     end
     script.puts "MAKEOPTS=(-j#{@options[:num_procs]})"
-    (ref_date, ref_time) = reference_datetime.split
-    script.puts "REFERENCE_DATETIME='#{reference_datetime}'"
-    script.puts "REFERENCE_DATE='#{ref_date}'"
-    script.puts "REFERENCE_TIME='#{ref_time}'"
     script.puts
+    (ref_datetime, ref_date, ref_time) = nil
     build_desc["remotes"].each do |remote|
       dir = sanitize(remote["dir"], remote["dir"])
+
+      author_date = `cd inputs/#{dir} && git log --format=@%at -1 | date +"%F %T" -u -f -`.strip
+      raise "error looking up author date in #{dir}" unless $?.exitstatus == 0
+      ref_datetime = build_desc["reference_datetime"] || author_date
```

Comment:
> in case there are no remotes.

I have added another commit to fail early, in case there are no remotes.

The other commits are squashed into one now.